### PR TITLE
Add user and password properties for Iceberg JDBC catalog

### PR DESCRIPTION
**Pull request by ebyhr**
## Description

Add user and password properties for Iceberg JDBC catalog to hide password
from the error message by Iceberg library. 

## Release notes

(x) Release notes are required, with the following suggested text:

```markdown
# Iceberg
* Add `iceberg.jdbc-catalog.connection-user` and `iceberg.jdbc-catalog.connection-password` config properties
  to JDBC catalog. ({issue}`issuenumber`)
```

### CODE REVIEW

File: plugin/trino-iceberg/src/test/java/io/trino/plugin/iceberg/catalog/jdbc/TestingIcebergJdbcServer.java [hunk 1]
```diff
@@ -64,12 +64,10 @@ public void execute(@Language("SQL") String sql)
     public String getJdbcUrl()
     {
         return format(
-                "jdbc:postgresql://%s:%s/%s?user=%s&password=%s&currentSchema=tpch,public",
+                "jdbc:postgresql://%s:%s/%s?currentSchema=tpch,public",
```
Comment (findinpath) [inline plugin/trino-iceberg/src/test/java/io/trino/plugin/iceberg/catalog/jdbc/TestingIcebergJdbcServer.java:67]: Why do we need `currentSchema=tpch,public` ?

Same question for `TestingJdbcServer` where after removing this string from the connection URL the tests were still 🟢

Comment (findinpath) [reply plugin/trino-iceberg/src/test/java/io/trino/plugin/iceberg/catalog/jdbc/TestingIcebergJdbcServer.java]: btw, this is also appearing in the product test environment configuration.

**Review by findinpath — APPROVED** [review]

**Review by phd3 — APPROVED** [review]

File: plugin/trino-iceberg/src/test/java/io/trino/plugin/iceberg/catalog/jdbc/TestingIcebergJdbcServer.java [hunk 2]
```diff
@@ -64,12 +64,10 @@ public void execute(@Language("SQL") String sql)
     public String getJdbcUrl()
     {
         return format(
-                "jdbc:postgresql://%s:%s/%s?user=%s&password=%s&currentSchema=tpch,public",
```
Comment (electrum) [inline plugin/trino-iceberg/src/test/java/io/trino/plugin/iceberg/catalog/jdbc/TestingIcebergJdbcServer.java:67]: Did you mean to remove the `currentSchema` property?

**Review by electrum — APPROVED** [review]

File: plugin/trino-iceberg/src/test/java/io/trino/plugin/iceberg/catalog/jdbc/TestingIcebergJdbcServer.java (hunk 2 above)
Comment (ebyhr) [reply plugin/trino-iceberg/src/test/java/io/trino/plugin/iceberg/catalog/jdbc/TestingIcebergJdbcServer.java:67]: Yes, exactly.

**Pull request merged**